tests/test_integration_crashpad.py

```diff
@@ -39,6 +39,49 @@ def test_crashpad_capture(cmake, httpserver):
     assert len(httpserver.log) == 2
 
 
+@pytest.mark.parametrize("port_correct", [True, False])
+def test_crashpad_crash_proxy_env(cmake, httpserver, port_correct):
+    if not shutil.which("mitmdump"):
+        pytest.skip("mitmdump is not installed")
+
+    proxy_process = None  # store the proxy process to terminate it later
+    port = "8080" if port_correct else "8081"
+    os.environ["http_proxy"] = f"http://localhost:{port}"
+    os.environ["https_proxy"] = f"http://localhost:{port}"
+    try:
+        proxy_process = start_mitmdump("http-proxy")
+
+        tmp_path = cmake(["sentry_example"], {"SENTRY_BACKEND": "crashpad"})
+
+        # make sure we are isolated from previous runs
+        shutil.rmtree(tmp_path / ".sentry-native", ignore_errors=True)
+
+        env = dict(os.environ, SENTRY_DSN=make_dsn(httpserver))
+        httpserver.expect_oneshot_request("/api/123456/minidump/").respond_with_data(
+            "OK"
+        )
+
+        try:
+            with httpserver.wait(timeout=10) as waiting:
+                child = run(tmp_path, "sentry_example", ["log", "crash"], env=env)
+                assert child.returncode  # well, it's a crash after all
+        except AssertionError as e:
+            if port_correct:
+                raise e
+            else:
+                assert len(httpserver.log) == 0
+                return
+
+        assert waiting.result
+        assert len(httpserver.log) == 1
```

Comment:
We currently can't know if the event was sent over the proxy, or if it used a fallback on the existing connection. It might make sense to listen to the port the proxy is on & confirm it passes through there.

On the other hand, the `port_correct` kind of already checks this by seeing if the example reads the proxy from the env (which is now 'wrong' and doesn't point to the running proxy server) and therefor no event ends up in the httpserver.

Reply:
After talking with you, I prefer to check the `mitmdump` output and count the logs there. We can do this at the end when `proxy_process` is terminated. This way, we don't have to guess whether another route by-passing the proxy allowed the request to succeed. This is also true for the other proxy tests.

Only thinking out aloud, maybe I missed a detail, but that would be the route I would choose first.